client/src/relay/updaters.tsx

```diff
@@ -39,7 +39,17 @@ function prependMessageToChannel(
     'Message',
   );
 
-  ConnectionHandler.insertEdgeBefore(messagesConnection, newMessageEdge);
+  // Insert new edge if the message is not already in connection.
+
+  const existingEdges = messagesConnection?.getLinkedRecords('edges') ?? [];
+
+  if (
+    !existingEdges.find(
+      (edge) =>
+        edge.getLinkedRecord('node')?.getDataID() === messageProxy.getDataID(),
+    )
+  )
+    ConnectionHandler.insertEdgeBefore(messagesConnection, newMessageEdge);
```

Comment:
Could you elaborate a bit on what does actually mean? `Insert new edge if the message is not already in connection.`?

Reply:
Because server does not filter the authenticated user's own message now, one message can be inserted twice into the Relay connection (one from mutation updater and another from subscription updater). This code is to check if there is already a node with the same ID before inserting a new message into the connection.

Reply:
Oh that sounds great!